pkg/tools/benchmark/job/service/job.go

```
@@ -186,20 +186,65 @@ func New(opts ...Option) (Job, error) {
 	if j.rps > 0 {
 		j.limiter = rate.NewLimiter(j.rps)
 	}
+	// If (Range.End - Range.Start) is smaller than Indexes, Indexes are prioritized based on Range.Start.
+	if (j.dataset.Range.End - j.dataset.Range.Start + 1) < j.dataset.Indexes {
+		j.dataset.Range.End = j.dataset.Range.Start + j.dataset.Indexes
+	}
+
 	return j, nil
 }
 
 func (j *job) PreStart(ctx context.Context) error {
-	log.Infof("[benchmark job] start download dataset of %s", j.hdf5.GetName().String())
-	if err := j.hdf5.Download(); err != nil {
-		return err
+	if j.jobType != GETOBJECT && j.jobType != EXISTS && j.jobType != REMOVE {
+		log.Infof("[benchmark job] start download dataset of %s", j.hdf5.GetName().String())
+		if err := j.hdf5.Download(j.dataset.URL); err != nil {
+			return err
+		}
+		log.Infof("[benchmark job] success download dataset of %s", j.hdf5.GetName().String())
+		log.Infof("[benchmark job] start load dataset of %s", j.hdf5.GetName().String())
+		var key hdf5.Hdf5Key
+		switch j.dataset.Group {
+		case "train":
+			key = hdf5.Train
+		case "test":
+			key = hdf5.Test
+		case "neighbors":
+			key = hdf5.Neighors
+		default:
```

Is it better to return an error when the dataset group is unknown?